chapter_06/rkc/exercise_3.pl

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/perl
+use warnings;
+use strict;
+
+# find the maximum key length for the given input hash
+sub key_max_length {
+	my %hash = @_;
```

Comment:
When you pass a hash as a parameter, perl actually passes in each key value pair as arguments, for example:

```perl
my %hash = ( a =>1, b =>2 );
some_function( %hash );
```
is equivalent to:
```perl
some_function( 'a', 1, 'b', 2 );
```
It's fine for the most part, but watch out when trying to pass large hashes like this.

Reply:
how would you recommend passing in large hashes? .. or is that covered in a later chapter?